x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -6,42 +6,36 @@
 
 import theme from '@elastic/eui/dist/eui_theme_light.json';
 import {
+  first,
   flatten,
   groupBy,
-  indexBy,
+  isEmpty,
   sortBy,
   uniq,
-  zipObject,
-  isEmpty,
-  first
+  zipObject
 } from 'lodash';
 import { TraceAPIResponse } from '../../../../../../../../server/lib/traces/get_trace';
 import { Span } from '../../../../../../../../typings/es_schemas/ui/Span';
 import { Transaction } from '../../../../../../../../typings/es_schemas/ui/Transaction';
 
-interface IWaterfallIndex {
-  [key: string]: IWaterfallItem | undefined;
-}
-
 interface IWaterfallGroup {
   [key: string]: IWaterfallItem[];
 }
 
 export interface IWaterfall {
-  entryTransaction?: Transaction;
-  traceRoot?: Transaction;
-  traceRootDuration?: number;
-
   /**
    * Duration in us
    */
   duration: number;
-  services: string[];
-  orderedItems: IWaterfallItem[];
-  itemsById: IWaterfallIndex;
-  getTransactionById: (id?: IWaterfallItem['id']) => Transaction | undefined;
-  errorCountByTransactionId: TraceAPIResponse['errorsPerTransaction'];
   serviceColors: IServiceColors;
+  entryTransaction?: Transaction;
+  rootTransaction?: {
+    item?: Transaction;
+    duration?: number;
```

Comment:
I think we don't need the `duration` separately. It was probably before we had optional chaining and this approach was easier if `rootTransaction` was undefined.

What I'm suggesting:

```ts
rootTransaction?: Transaction
```

Reply:
Then to get the duration `waterfall.rootTransaction?.transaction.duration.us`